databricks-sdk-java/src/main/java/com/databricks/sdk/core/DefaultCredentialsProvider.java

```diff
@@ -80,4 +93,70 @@ public synchronized HeaderFactory configure(DatabricksConfig config) {
             + authFlowUrl
             + " to configure credentials for your preferred authentication method");
   }
+
+  /**
+   * Adds OpenID Connect (OIDC) based credential providers to the list of available providers.
+   *
+   * @param config The Databricks configuration containing OIDC settings
+   */
+  private void addOIDCCredentialsProviders(DatabricksConfig config) {
+    OpenIDConnectEndpoints endpoints = null;
+    try {
+      endpoints = config.getOidcEndpoints();
+    } catch (Exception e) {
+      LOG.warn("Failed to get OpenID Connect endpoints", e);
+    }
```

Comment:
When I read this code the first time, I got concerned that we might end-up with an NullPointerException downstream. It might be worth throwing an IllegalArgumentException right here so that the intent is clear. This is also less bugprone as it would completely prevent potentials issue if someone change the code to use `endpoints` before the `DatabricksOAuthTokenSource` validation.

```suggestion
    OpenIDConnectEndpoints endpoints;
    try {
      endpoints = config.getOidcEndpoints();
    } catch (Exception e) {
      LOG.warn("Failed to get OpenID Connect endpoints", e);
      throw e;
    }
```

Reply:
From my understanding, we aim to instantiate all providers regardless of whether they are correctly configured, and only fail when a token is actually requested. Throwing an exception during instantiation would prevent OIDC providers from being created. Instead, we could return the exception as you mentioned or, as done in the Go SDK, pass the endpoint getter to the token source. @hectorcast-db suggested that using a try-catch block at this stage might be preferable, but we can discuss the best approach.

Reply:
We definitely not want to throw an error here, because any user which has OIDC disabled in their workspace would be blocked from using the SDKs. We have two options:
* Skip the creation of the CredentialsStrategy
* Create the CredentialsStrategy with a null parameter

I think the second option is more inlined with the rest of the SDKs: We create all CredentialsStrategy even if some parameters are missing and do the validation at `Configure` time. 
This also impacts the logs. When configuring, users will see the `skipping xxx auth` and possible error logs for those which are skipped. If we don't create the CredentialsStrategy at all, the user wont' have the logs (or will have them in a different place if we log it here where the error is thrown).

But I am not strongly  opposed to skip the creation since we are planning to rethink DefaultAuth for SDK Mod.

Reply:
Closing this thread as per the comment.